hibernate-core/src/main/java/org/hibernate/graph/internal/Helper.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or http://www.gnu.org/licenses/lgpl-2.1.html
+ */
+package org.hibernate.graph.internal;
+
+import java.util.Map;
+import javax.persistence.metamodel.Attribute;
+
+import org.hibernate.graph.SubGraph;
+import org.hibernate.graph.spi.AttributeNodeImplementor;
+import org.hibernate.graph.spi.SubGraphImplementor;
+import org.hibernate.metamodel.model.domain.spi.AttributeImplementor;
+import org.hibernate.metamodel.model.domain.spi.IdentifiableTypeImplementor;
+import org.hibernate.metamodel.model.domain.spi.MapAttributeImplementor;
+import org.hibernate.metamodel.model.domain.spi.PluralAttributeImplementor;
+import org.hibernate.metamodel.model.domain.spi.SimpleTypeImplementor;
+import org.hibernate.metamodel.model.domain.spi.SingularAttributeImplementor;
+
+/**
+ * @author Steve Ebersole
+ */
+public class Helper {
```

Comment:
Do we really want a public class with such a generic name?

Reply:
internal is not a "public class"

Reply:
Note - I am not saying the class name is great, just pointing out that there is a difference between "public" as in:

1. Java keyword
2. What we support applications using

This does not fit the later - that's why it is in an internal package

Reply:
Yeah, I was referring as `public` in the Java sense e.g. a class that will be visible from all the other classes when I develop in ORM.

`GraphHelper` or `EntityGraphHelper` would be better IMHO.

Reply:
You mean because internal classes only ever get used from the package they are defined in? ;)

Look, I already said the name could be improved.  In early work a name (*especially* for an internal utility class) is utterly unimportant to me.  `GraphHelper` - great.

But I utterly disagree with this "it should be private / package-protected" stance.